### DISCUSSION TITLE
Where is daemon.json file on MacOS

### DISCUSSION THREAD

**Discussion by a4z** [rancher-sandbox/rancher-desktop#3371] [Q&A]
Rancher-desktop MacOS, 
I need to add

```
{
  "insecure-registries" : ["mylocalhost:5000"]
}
```
into daemon.json,  but I can't find the file. 

Where is it on MacOS / Rancher-Desktop / Moby ?

**Comment by guzelcihad** [comment]
Hi,
It's in the directory on your host `.docker/daemon.json`. But adding insecure-registries to this file doesn't solve my problem. People recommend to use this solution -> [discussion](https://github.com/rancher-sandbox/rancher-desktop/discussions/1477)
It works for me

**Reply by eunrui** [reply]
Thanks, I found onther way to edit daemon.json for colima by this link.
If anyone need help and use colima, follow me.
```
colima start --edit 
```
(find docker:  and edit it )
```
docker:
  insecure-registries:
    - ip:port
```
then apply changes
```
colima ssh sudo systemctl daemon-reload
colima ssh sudo systemctl restart docker
```

**Comment by bneradt** [comment]
I used to be able to follow the instructions https://github.com/rancher-sandbox/rancher-desktop/discussions/1477#discussioncomment-2106389, but restarting Rancher now resets the files I modify in the Rancher image. Both the `/etc/conf.d/docker` and the `/etc/docker/daemon.json` files get rewritten to their original values after I restart Rancher through the UI. Is there a way to get these files to persist after restarting Rancher. I'm running Rancher Version 1.19.3. 

I wonder whether it is related to this `lima.log` entry:

```
2025-07-13T01:48:56.577Z: > /Applications/Rancher Desktop.app/Contents/Resources/resources/darwin/lima/bin/limactl shell --workdir=. 0 sudo mv /tmp/rd-docker-JwMEqF.docker /etc/conf.d/docker
```
